evalml/tests/component_tests/test_featuretools.py

```diff
@@ -163,3 +164,140 @@ def test_dfs_sets_max_depth_1(mock_dfs, X_y_multi):
     feature.fit(X_pd, y)
     _, kwargs = mock_dfs.call_args
     assert kwargs["max_depth"] == 1
+
+
+@patch("evalml.pipelines.components.transformers.preprocessing.featuretools.dfs")
+def test_dfs_with_serialized_features(mock_dfs, X_y_binary):
+    X, y = X_y_binary
+    X_pd = pd.DataFrame(X)
+    X_pd.columns = X_pd.columns.astype(str)
+
+    es = ft.EntitySet()
+    es = es.add_dataframe(
+        dataframe_name="X", dataframe=X_pd, index="index", make_index=True
+    )
+    feature_matrix, features = ft.dfs(
```

Comment:
I'm not seeing the same behavior with the `year` primitive on the fraud dataset:

```python
from evalml.demos import load_fraud
import featuretools as ft
from evalml.pipelines.components import DFSTransformer
from featuretools.feature_base import IdentityFeature
import pandas as pd
import pytest


X, y = load_fraud(1000)
del X.ww

X_fit = X.iloc[:X.shape[0]//2]
X_transform = X.iloc[X.shape[0]//2:]


es = ft.EntitySet()
es = es.add_dataframe(
    dataframe_name="X", dataframe=X_fit, index="index", make_index=True
)
feature_matrix, features = ft.dfs(
    entityset=es, target_dataframe_name="X", trans_primitives=["year"]
)
features = list(filter(lambda f: not isinstance(f, IdentityFeature), features))

dfs = DFSTransformer(features=features)
dfs.fit(X_fit)
X_t = dfs.transform(X_transform)

with pytest.raises(AssertionError):
    pd.testing.assert_frame_equal(X_t, feature_matrix)
    
print(X_t.columns)
```

I would expect the same behavior in this test for this repro: X_t has all the original features plus YEAR (and is therefore equal to feature_matrix).

Reply:
This was my mistake heh. We need to pass the identity features in this case for X_t to match feature_matrix.